wolfcrypt/src/port/nxp/dcp_port.c

```diff
@@ -0,0 +1,537 @@
+    aes->handle.keySlot = dcp_key_slot(aes->handle.channel);
+    aes->handle.swapConfig = kDCP_NoSwap;
+    return 0;
+}
+
+void DCPAesFree(Aes *aes)
+{
+    dcp_free(aes->handle.channel);
+}
+
+int  DCPAesSetKey(Aes* aes, const byte* key, word32 len, const byte* iv,
+                          int dir)
+{
+
+#if DCP_USE_OTP_KEY
+#warning Please update cipherAes128 variables to match expected AES ciphertext for your OTP key.
+#endif
+    status_t status;
+    if (!aes || !key)
+        return BAD_FUNC_ARG;
+
+    if (len != 16)
+        return BAD_FUNC_ARG;
+    if (aes->handle.channel == 0) {
+        return BAD_FUNC_ARG;
+    }
+    dcp_lock();
+    status = DCP_AES_SetKey(DCP, &aes->handle, key, 16);
+    if (status != kStatus_Success)
+        status = WC_HW_E;
+    else {
+        if (iv)
+            XMEMCPY(aes->reg, iv, 16);
+        else
+            XMEMSET(aes->reg, 0, 16);
+    }
+    dcp_unlock();
+    return status;
+}
+
+int  DCPAesCbcEncrypt(Aes* aes, byte* out, const byte* in, word32 sz)
+{
+    int ret;
+    if (sz % 16)
+        return BAD_FUNC_ARG;
+    dcp_lock();
+    ret = DCP_AES_EncryptCbc(DCP, &aes->handle, in, out, sz, (const byte *)aes->reg);
+    if (ret)
+        ret = WC_HW_E;
+    else
+        XMEMCPY(aes->reg, out, AES_BLOCK_SIZE);
+    dcp_unlock();
+    return ret;
+}
+
+int  DCPAesCbcDecrypt(Aes* aes, byte* out, const byte* in, word32 sz)
+{
+    int ret;
+    if (sz % 16)
+        return BAD_FUNC_ARG;
+    dcp_lock();
+    ret = DCP_AES_DecryptCbc(DCP, &aes->handle, in, out, sz, (const byte *)aes->reg);
+    if (ret)
+        ret = WC_HW_E;
+    else
+        XMEMCPY(aes->reg, in, AES_BLOCK_SIZE);
+    dcp_unlock();
+    return ret;
+}
+
+int  DCPAesEcbEncrypt(Aes* aes, byte* out, const byte* in, word32 sz)
+{
+    int ret;
+    if (sz % 16)
+        return BAD_FUNC_ARG;
+    dcp_lock();
+    ret = DCP_AES_EncryptEcb(DCP, &aes->handle, in, out, sz);
+    if (ret)
+        ret = WC_HW_E;
+    dcp_unlock();
+    return ret;
+}
+
+int  DCPAesEcbDecrypt(Aes* aes, byte* out, const byte* in, word32 sz)
+{
+    int ret;
+    if (sz % 16)
+        return BAD_FUNC_ARG;
+    dcp_lock();
+    ret = DCP_AES_DecryptEcb(DCP, &aes->handle, in, out, sz);
+    if (ret)
+        ret = WC_HW_E;
+    dcp_unlock();
+    return ret;
+}
+
+#endif
+
+#ifndef NO_SHA256
+int wc_InitSha256_ex(wc_Sha256* sha256, void* heap, int devId)
+{
+    int ret;
+    int ch;
+    int keyslot;
+    if (sha256 == NULL)
+        return BAD_FUNC_ARG;
+    ch = dcp_get_channel();
+    if (ch == 0)
+        return WC_PENDING_E;
+    keyslot = dcp_key_slot(ch);
+
+    dcp_lock();
+    (void)devId;
+    XMEMSET(sha256, 0, sizeof(wc_Sha256));
+    sha256->handle.channel    = ch;
+    sha256->handle.keySlot    = keyslot;
+    sha256->handle.swapConfig = kDCP_NoSwap;
+    ret = DCP_HASH_Init(DCP, &sha256->handle, &sha256->ctx, kDCP_Sha256);
+    if (ret != kStatus_Success)
+        ret = WC_HW_E;
+    dcp_unlock();
+
+    return ret;
+}
```

Comment:
```suggestion
int wc_InitSha256_ex(wc_Sha256* sha256, void* heap, int devId)
{
    int ret;
    int ch;
    if (sha256 == NULL)
        return BAD_FUNC_ARG;
    ch = dcp_get_channel();
    if (ch == 0)
        return WC_PENDING_E;

    dcp_lock();
    (void)devId;
    XMEMSET(sha256, 0, sizeof(wc_Sha256));
    sha256->handle.channel    = ch;
    sha256->handle.keySlot    = kDCP_KeySlot0;
    sha256->handle.swapConfig = kDCP_NoSwap;
    ret = DCP_HASH_Init(DCP, &sha256->handle, &sha256->ctx, kDCP_Sha256);
    if (ret != kStatus_Success)
        ret = WC_HW_E;
    dcp_unlock();

    return ret;
}
```
Key slot is not used in hash, just set to default value